### ISSUE TITLE
Nudity detection for videos

### ISSUE THREAD

**Issue by lfcnassif** [sepinf-inc/IPED#129]
Currently just pictures are sent to nudity detection module. We could simply send current extracted video frames for detection, could pick up higher score, score mean, or store all scores as multivalued metadata.

Number of video frames to be inspected also could be changed from fixed to dynamic in configuration, proportial to video duration.

**Comment by tc-wleite** [comment]
As an initial implementation, I am calling the current explicit image detection algorithm (DIE) for each frame already extracted by the VideoThumbTask (currently 20 in the default configuration), so it will depend on the video task (which I believe is fine).
With this setup (calling DIE), I do not believe more frames would help.
Obviously, the results can be improved a lot, from my initial tests, this initial simple solution can still be useful.
I am also retraining the model, to include some images that came from videos (not only "pure" images as before).

**Comment by lfcnassif** [comment]
For sure, it will be very useful and will help a lot! Thank you! Will you store the greatest detected nudity score?

About extracting more frames, I think evaluating on a test set could say if it will improve or not, but it could be done later, don't worry.

Retraining with video frames is a good ideia! But will need much more work on your side...

**Comment by tc-wleite** [comment]
I am using a weighted average to combine frames scores into a single score, with higher weights for the frames with higher scores. 
That seems to produce much better results than just using the top score. And it is a very simplified way of update probabilities based on multiple independent (which is not exactly the case) observations (Bayes' theorem).

About retraining the model, I spent some time selecting training video frames, but using good sources to collect the videos, the current detection algorithm and the image similarity algorithm to discard similar frames, made the process relatively fast.
Besides that, my goal (already achieved) was to add a small amount of such images (~10% of the existing training set), so the current detection won't be sacrificed.
I am also using this opportunity to reduce the model file length a bit.
By the way, with the new workstation I am using, training is much faster, which allows more attempts with different setups. I already have a new model and plan to train a few more during the next days.

**Comment by lfcnassif** [comment]
Excellent!
